File: C/alogrithm/sort/quick_sort.c
#include <stdio.h>
int a[101], t;

void quick_sort(int left, int right)
{
	if(left > right)
		return;

	int i = left, j = right, pivot = a[left], temp;

	while(i != j)
	{
		while(a[j] >= pivot && i < j)
			j--;
		while(a[i] <= pivot && i < j)
			i++;

		if(i < j)
		{
			temp = a[i]; a[i] = a[j]; a[j] = temp;
		}
	}

	t = a[left]; a[left] = a[i]; a[i] = t;

	quick_sort(left, i-1);
	quick_sort(i+1, right);

}

int main(int argc, char const *argv[])
{
	int i,n;
	// 输入
	scanf("%d",&n);
	for(i = 0;i<n;i++)
		scanf("%d",&a[i]);
	// 调用以数组下标为准
	quick_sort(0,n-1);
	// 输出
	for(i=0;i<n;i++)
		printf("%d", a[i]);
	return 0;
}

File: Python3/algorithm/Sort/quick_sort.py
def q_srot(a, left, right):
	# 递归出口
	if left>right:
		return

	i=left
	j=right
	pivot=a[left]

	while i!=j:
		while a[j]>pivot and i<j:
			j-=1
		while a[i]<=pivot and i<j:#由于第一个元素就是其本身，而本身直接不需要进行交换（下标要略过）所以要加上等号。
			i+=1

		if i<j:
			a[i], a[j]=a[j], a[i]

	a[left], a[i]=a[i], a[left] #数字归位

	q_srot(a, left, i-1)
	q_srot(a, i+1, right)

a = [int(n) for n in input().split(',')]

q_srot(a, 0, len(a)-1)

for a in a:
	print(a)